plenum/server/node.py

```diff
@@ -739,31 +775,37 @@ def onConnsChanged(self, joined: Set[str], left: Set[str]):
                 self.status = Status.starting
         self.elector.nodeCount = self.connectedNodeCount
 
-        if self.master_primary in joined:
+        if self.master_primary_name in joined:
             self.lost_primary_at = None
-        if self.master_primary in left:
+        if self.master_primary_name in left:
             logger.debug('{} lost connection to primary of master'.format(self))
             self.lost_master_primary()
-
         if self.isReady():
             self.checkInstances()
+
+
             # TODO: Should we only send election messages when lagged or
             # otherwise too?
-            if isinstance(self.elector, PrimaryElector) and joined:
-                msgs = self.elector.getElectionMsgsForLaggedNodes()
-                logger.debug("{} has msgs {} for new nodes {}".
-                             format(self, msgs, joined))
-                for joinedNode in joined:
-                    self.sendElectionMsgsToLaggingNode(joinedNode, msgs)
-                    # Communicate current view number if any view change
-                    # happened to the connected node
-                    if self.viewNo > 0:
-                        logger.debug("{} communicating view number {} to {}"
-                                     .format(self, self.viewNo-1, joinedNode))
-                        rid = self.nodestack.getRemote(joinedNode).uid
-                        self.send(
-                            self._create_instance_change_msg(self.viewNo, 0),
-                            rid)
+            # if isinstance(self.elector, PrimaryElector) and joined:
```

Comment:
This code needs to be moved to PrimaryElector level (actually we need to have it in PrimaryDecider interface)